src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight2B2T.kt

```diff
@@ -0,0 +1,346 @@
+                reset()
+                return@safeListener
+            }
+
+            /* If we are not wearing an elytra then reset */
+            if (player.inventory.armorInventory[2].item != Items.ELYTRA && started != MovementState.NOT_STARTED) {
+                reset()
+                return@safeListener
+            }
+
+            when (started) {
+                MovementState.NOT_STARTED -> {
+                    calcStartingHeight()
+                }
+                MovementState.IDLE -> {
+                    val targetPos = getNextIdle()
+                    player.setVelocity(targetPos.x, targetPos.y, targetPos.z)
+                }
+                MovementState.MOVING -> {}
+            }
+        }
+
+        /**
+         * Listen to PlayerTravelEvent in the case where the user is pressing the spacebar and we have not yet started.
+         * This allows us to determine if we need to handle a jumping or a falling start.
+         *
+         * Also cancel movement that would otherwise cause us to jitter in the -Y direction
+         */
+        safeListener<PlayerTravelEvent> {
+            /* If are are pressing the space and not started then we need to start in the jump state */
+            if (player.movementInput.jump) {
+                if (started == MovementState.NOT_STARTED) {
+                    isJumpStart = true
+                    return@safeListener
+                } else {
+                    /* Don't go up but maintain other velocities if the user tries to fly up */
+                    player.setVelocity(player.motionX, 0.0, player.motionZ)
+                    /* In case they double press space (exits flying mode normally) */
+                    player.capabilities.isFlying = true
+                }
+            }
+
+            if (player.movementInput.sneak && started != MovementState.NOT_STARTED) {
+                player.setVelocity(0.0, -descendSpeed.value, 0.0)
+                return@safeListener
+            }
+
+            when (started) {
+                MovementState.NOT_STARTED -> {}
+                MovementState.IDLE -> {
+                    if (isInputting) {
+                        started = MovementState.MOVING
+                        accelStart = System.currentTimeMillis()
+                    }
+                }
+                MovementState.MOVING -> {
+                    if (isInputting) {
+                        if (player.speed < maxVelocity.value) {
+                            val accelVec = player.lookVec.scale(((System.currentTimeMillis() - accelStart) / (10000 - (accelerateSpeed.value * 10000 - 1))).toDouble())
+                            player.setVelocity(accelVec.x, 0.0, accelVec.z)
+                        } else {
+                            player.setVelocity(player.motionX, 0.0, player.motionZ)
+                            it.cancel()
+                        }
```

Comment:
MovementUtils has a function for calculating movement yaw from player input. Use that with some trigonometric math